plugins/web/api/api.js

```diff
@@ -1,123 +1,108 @@
-                    params.qstring.metrics._device = (agent.device.family === "Other") ? "Unknown" : agent.device.family;
-                }
+            //if some metrics are not provided, parse them from user agent                
+            if (!params.qstring.metrics._browser) {
+                params.qstring.metrics._browser = agent.browser.name;
             }
 
-            //check if view events need to have platform segment
-            if (params.qstring.events && params.qstring.events.length && Array.isArray(params.qstring.events)) {
-                params.qstring.events = params.qstring.events.map(function(currEvent) {
-                    if (currEvent.key === "[CLY]_view" && currEvent.segmentation && currEvent.segmentation.name && !currEvent.segmentation.segment) {
-                        currEvent.segmentation.segment = data.os;
-                    }
-                    return currEvent;
-                });
+            if (!params.qstring.metrics._browser_version) {
+                params.qstring.metrics._browser_version = agent.browser.version;
             }
 
-            //check of any crash segments can be updated
-            if (typeof params.qstring.crash === "string") {
-                try {
-                    params.qstring.crash = JSON.parse(params.qstring.crash);
-                }
-                catch (SyntaxError) {
-                    console.log('Parse crash JSON failed');
-                    return false;
-                }
-                if (!params.qstring.crash._os) {
-                    params.qstring.crash._os = data.os;
-                }
+            if (params.qstring.metrics._browser && params.qstring.metrics._browser_version && !params.qstring.metrics._browser_version.startsWith("[" + params.qstring.metrics._browser.toLowerCase() + "]_")) {
+                params.qstring.metrics._browser_version = "[" + params.qstring.metrics._browser.toLowerCase() + "]_" + params.qstring.metrics._browser_version;
+            }
 
-                if (!params.qstring.crash._os_version) {
-                    params.qstring.crash._os_version = data.os + " " + data.os_version;
-                }
+            if (!params.qstring.metrics._os) {
+                params.qstring.metrics._os = data.os;
+            }
 
-                if (!params.qstring.crash._browser) {
-                    params.qstring.crash._browser = agent.toAgent();
-                }
+            if (!params.qstring.metrics._os_version) {
+                params.qstring.metrics._os_version = data.os_version;
+            }
 
-                if (!params.qstring.crash._device) {
-                    params.qstring.crash._device = (agent.device.family === "Other") ? "Unknown" : agent.device.family;
-                }
+            if (!params.qstring.metrics._device) {
+                params.qstring.metrics._device = (agent.device.vendor === "Other") ? "Unknown" : agent.device.vendor;
```

Comment:
For device can we do model and fallback to Vendor if model is not available?
Same for crashes too.